### PR TITLE
Kadane's Algorithm in js

### PR DESCRIPTION
**Pull request by dikshajoshi18**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

the sum of contiguous subarray within a one-dimensional array of numbers that has the largest sum. 

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

issue = #5135

### CODE REVIEW

File: JavaScript/Algorithms/kadane's_algo.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    } 
+}
+
+
+var arr = [-2, 1, -3, 4, -1, 2, 1, -5, 4];
+let obj = new Solution();
+let ans = obj.kadanesalgo(arr);
+if(ans===0)
+    ans=0;
+console.log(ans);
```
Comment (siddharth25pandey) [inline JavaScript/Algorithms/kadane's_algo.js]: Take Dynamic input & display relevant message regarding I/O

File: JavaScript/Algorithms/kadane's_algo.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+// solved example
+// Input:
+// N = 4
+// arr[] = {-1,-2,-3,-4}
+// Output:
+// -1
+// Explanation:
+// Max subarray sum is -1 
+// of element (-1)
```
Comment (siddharth25pandey) [inline JavaScript/Algorithms/kadane's_algo.js:76]: Add Time & Space Complexity

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@dikshajoshi18 Do the requested changes

File: JavaScript/Algorithms/kadane's_algo.js (hunk 2 above)
Comment (dikshajoshi18) [reply JavaScript/Algorithms/kadane's_algo.js:76]: I updated according to the required changes

File: JavaScript/Algorithms/kadane's_algo.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+
+// Input:
+// N = 4
+// arr[] = {-1,-2,-3,-4}
+// Output:
+// -1
```
Comment (siddharth25pandey) [inline JavaScript/Algorithms/kadane's_algo.js]: Sample I/O is not matching with the execution of the Program.
**Display message is missing**

File: JavaScript/Algorithms/kadane's_algo.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+// solved example
+// Input:
+// N = 4
+// arr[] = {-1,-2,-3,-4}
+// Output:
+// -1
+// Explanation:
+// Max subarray sum is -1 
+// of element (-1)
```
Comment (siddharth25pandey) [inline JavaScript/Algorithms/kadane's_algo.js]: Remove the Redundant lines, Already present in bottom of the code

File: JavaScript/Algorithms/kadane's_algo.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        console.log("INPUT ARRAY ELEMENTS");
+        let a = new Array(ele);
+        for(let k=0;k<ele;k++){
+            a[k]=line[k];
+        }
+        let Ob = new Kadanes();
+        let sol = Ob.kadanesalgo(a,ele);
+        if(sol===0)
+            sol=0;
+        console.log(sol);
```
Comment (siddharth25pandey) [inline JavaScript/Algorithms/kadane's_algo.js:45]: Display a message in print form as what are you showing output to the user

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]
@dikshajoshi18 Small changes are required

File: JavaScript/README.md [hunk 6]
```diff
@@ -16,6 +16,7 @@
 - [Boyer Moore's Majority Vote Algorithm](Algorithms/BoyerMooreAlgorithm.js)
 - [Armstrong Number](Algorithms/Armstrong.js)
 - [Longest Increaing Subsequnce](Algorithms/LongestIncreasingSubsequenceDP.js)
+- [Maximum Sum of Subarray](Algorithms/kadane's_algo.js)
```
Comment (ankitaggarwal23) [inline JavaScript/README.md]: rename it ` [Maximum Sum of Subarray]` to `Kadane's Algorithm`

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: JavaScript/README.md (hunk 6 above)
Comment (dikshajoshi18) [reply JavaScript/README.md]: done please check it

File: JavaScript/Algorithms/kadane's_algo.js (hunk 5 above)
Comment (dikshajoshi18) [reply JavaScript/Algorithms/kadane's_algo.js:45]: done please check it

**Review by siddharth25pandey — APPROVED** [review]
@dikshajoshi18 LGTM 💯 Able to merge

**Comment by siddharth25pandey** [comment]
@all-contributors please add @dikshajoshi18 for her code

**Pull request merged**